services/provider/server/server.go

```diff
@@ -278,23 +278,29 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
 	}
 
-	// Service for monitoring endpoints
-	scMonitoring := &v1.Service{}
-	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	// Get mgr pod hostIP
+	podList := &corev1.PodList{}
+	err = s.client.List(ctx, podList, client.MatchingLabels(map[string]string{"app": "rook-ceph-mgr"}))
+	if err != nil || len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to list pod with rook-ceph-mgr label. %v", err)
 	}
 
```

Comment:
You are going to reference the first pod a couple of times, It will be preferable to take a pointer for it and use it instead of `podList.Items[0]`

```
mgrPod := &podList.Items[0]
```